// zksync/build/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.wallet = exports.crypto = exports.utils = exports.types = exports.getDefaultProvider = exports.closestPackableTransactionAmount = exports.closestPackableTransactionFee = exports.ETHProxy = exports.Provider = exports.Signer = exports.Wallet = void 0;
const wallet_1 = require("./wallet");
Object.defineProperty(exports, "Wallet", { enumerable: true, get: function () { return wallet_1.Wallet; } });
const provider_1 = require("./provider");
Object.defineProperty(exports, "Provider", { enumerable: true, get: function () { return provider_1.Provider; } });
Object.defineProperty(exports, "ETHProxy", { enumerable: true, get: function () { return provider_1.ETHProxy; } });
Object.defineProperty(exports, "getDefaultProvider", { enumerable: true, get: function () { return provider_1.getDefaultProvider; } });
const signer_1 = require("./signer");
Object.defineProperty(exports, "Signer", { enumerable: true, get: function () { return signer_1.Signer; } });
const utils_1 = require("./utils");
Object.defineProperty(exports, "closestPackableTransactionAmount", { enumerable: true, get: function () { return utils_1.closestPackableTransactionAmount; } });
Object.defineProperty(exports, "closestPackableTransactionFee", { enumerable: true, get: function () { return utils_1.closestPackableTransactionFee; } });
const wallet = __importStar(require("./wallet"));
exports.wallet = wallet;
const types = __importStar(require("./types"));
exports.types = types;
const utils = __importStar(require("./utils"));
exports.utils = utils;
const crypto = __importStar(require("./crypto"));
exports.crypto = crypto;
